src/main/java/com/refinedmods/refinedstorage/screen/grid/GridScreen.java

```diff
@@ -128,12 +140,24 @@ public void onPostInit(int x, int y) {
         addSideButton(new GridSizeSideButton(this, () -> grid.getSize(), size -> grid.onSizeChanged(size)));
 
         if (grid.getGridType() == GridType.PATTERN) {
+
+            patternScrollbar = new ScrollbarWidget(this, 160, getTopHeight() + getVisibleRows() * 18 + 4, 6, 18 * 3 - 2, true);
+            patternScrollOffsetAbsoluteMax = GridNetworkNode.processingMatrixSize / 3 - 3;
+            patternScrollbar.setMaxOffset(patternScrollOffsetAbsoluteMax);
+            patternScrollbar.setOffset(patternScrollOffset); // keep offset when changing between fluid and item view
+            container.updatePatternSlotPositions(patternScrollOffset);
+            patternScrollbar.addListener((oldOffset, newOffset) -> {
+                patternScrollOffset = newOffset;
+                container.updatePatternSlotPositions(newOffset);
+            });
```

Comment:
This block of code needs some more newlines here and there. It's quite unreadable now.